core/templates/pages/exploration-editor-page/changes-in-human-readable-form/changes-in-human-readable-form.component.html

```diff
@@ -147,8 +103,136 @@
+                <i>Explanation: </i>
+                <div class="explanation" [innerHTML]="lostChange.newValue.explanation.html">
+                </div>
+              </div>
+            </div>
+
+            <div *ngIf="lostChange.isNewValueEmpty()" class="state-edit-desc">
+              <span>Removed solution</span>
+            </div>
+          </div>
+
         </ng-container>
       </div>
+      <div *ngIf="lostChange.cmd === 'edit_exploration_property'">
+        <ng-container [ngSwitch]="lostChange.propertyName">
+          <div *ngSwitchCase="'title'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New title: </strong>{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed title</strong>
+            </div>
+          </div>
+          <div *ngSwitchCase="'category'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New category: </strong>{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed category</strong>
+            </div>
+          </div>
+          <div *ngSwitchCase="'objective'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New objective: </strong>{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed objective</strong>
+            </div>
+          </div>
+          <div *ngSwitchCase="'language_code'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New language code: </strong >{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed language code</strong>
+            </div>
+          </div>
+          <div *ngSwitchCase="'init_state_name'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New initiating state name: </strong>{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed initiating state name</strong>
+            </div>
+          </div>
+          <div *ngSwitchCase="'tags'">
+            <div *ngIf="!lostChange.isNewValueEmpty()">
+              <strong>New tags: </strong>{{lostChange.newValue}}
+            </div>
+            <div *ngIf="lostChange.isNewValueEmpty()">
+              <strong>Removed tags</strong>
+            </div>
+          </div>
+        </ng-container>
+      </div>
+      <div *ngIf="lostChange.cmd === 'add_written_translation'">
+        <strong>Change in translation</strong>
+        <div class="sub-edit">
+          <div class="language_code">
+            <i>Language Code: </i> {{lostChange.languageCode}}
+          </div>
+          <div class="content_id">
+            <i>Content id: </i> {{lostChange.contentId}}
```

Comment:
Content id might not make sense to a user since it is an internal field, right?

Reply:
But content-id usually contains the property which is changed, so it can give the idea to the user. Otherwise, how we'll show the user that this content translation is added or changed.

Reply:
Need not show that the content id changed. When there is an update to the id, maybe we can show the corresponding content instead.

Reply:
> But content-id usually contains the property which is changed, so it can give the idea to the user. Otherwise, how we'll show the user that this property translation is added or changed.

Here "property which is changed" implies the property for which translation is added or changed. I guess I didn't explain it properly in the previous comment. Sorry for this.
But here we don't have the content. We only have the content id for which the translation is added or deleted. So if we'll not show the content id then the user will never know that for which property this translation was. I hope it's clear now.